### ISSUE TITLE
On computation of vae_loss

### ISSUE THREAD

**Issue by chandu8542** [wiseodd/generative-models#34]
Hi,

Can you tell me on thing:  Are recon_loss and kl_loss in vae tensorflow script  scalar or vector?

**Comment by wiseodd** [comment]
They're vectors.

**Comment by chandu8542** [comment]
How is that possible since if X is a matrix and logit is a vector, cross entropy will not work (as per tf doc, incompitability). If X is a vector and logit is also a vector, recon_loss will be a scalar due to reduce_sum and hence recon_loss will be a vector is ruled out. Third possiblity is that logit must be a matrix which is not the case. Hence, can you give me a concrete calculation?

**Comment by wiseodd** [comment]
Hi,

`X` is a matrix and `logits` is also a matrix. The cross entropy matrix is then marginalized column wise. Therefore `recon_loss` is a column vector. Same with `kl_loss`.

I suggest you run the code and print them. It should be clear then.
